tests/rpm-ostree/main.yml

```diff
@@ -0,0 +1,571 @@
+      register: current_dir
+
+    - name: Get json output from rpm-ostree status
+      command: rpm-ostree status --json
+      register: ros_status
+
+    - name: Convert to json
+      set_fact:
+         ross_json: "{{ ros_status.stdout | from_json }}"
+
+    - name: Set current commit version
+      set_fact:
+         head_csum: "{{ ross_json['deployments'][0]['checksum'] if ross_json['deployments'][0]['booted'] else ross_json['deployments'][1]['checksum'] }}"
+
+    - name: Create local branch
+      command: ostree refs --create local-branch {{ head_csum }}
+
+    - name: Update origin file
+      command: sed -i 's/^\(.*refspec\)=.*$/\1=local-branch/g' {{ current_dir.stdout }}.origin
+
+    - name: Print file contents
+      command: cat {{ current_dir.stdout }}.origin
+
+    - name: Commit new local branch
+      command: ostree commit -b local-branch --tree=ref=local-branch --add-metadata-string version=test
+      register: new_commit
+
+    - include: roles/rpm_ostree_upgrade/tasks/main.yml
+
+    - include: ../../common/ans_reboot.yml
+
+    - include: roles/rpm_ostree_status_verify/tasks/main.yml
+      booted_checksum: "{{ new_commit.stdout }}"
+      booted_version: "test"
+
+    - name: Rebase back to original deployment
+      command: rpm-ostree rebase {{ refspec.stdout }} {{ head_csum }}
+
+    - include: ../../common/ans_reboot.yml
+
+    - include: roles/rpm_ostree_status_verify/tasks/main.yml
+      booted_checksum: "{{ head_csum }}"
+
+    - name: Cleanup
+      command: rpm-ostree cleanup -rpmb
+
+#####################################################################################
+#
+# rpm-ostree upgrade + rebase + install + uninstall
+#   - simulates an upgrade and rebases to origin with install/uninstall
+#
+#####################################################################################
+
+- name: rpm-ostree - upgrade and rebase + install and uninstall
+  hosts: all
+  become: yes
+
+  tags:
+    - upgrade_rebase_install_uninstall
+
+  vars_files:
+    - vars.yml
+
+  tasks:
+    - name: Get current refspec
+      shell: cat $(ostree admin --print-current-dir).origin | grep refspec | cut -d= -f2
+      register: refspec
+
+    - name: Set origin file
+      command: ostree admin --print-current-dir
+      register: current_dir
+
+    - name: Get json output from rpm-ostree status
+      command: rpm-ostree status --json
+      register: ros_status
+
+    - name: Convert to json
+      set_fact:
+         ross_json: "{{ ros_status.stdout | from_json }}"
+
+    - name: Set current commit version
+      set_fact:
+         head_csum: "{{ ross_json['deployments'][0]['checksum'] if ross_json['deployments'][0]['booted'] else ross_json['deployments'][1]['checksum'] }}"
+
+    - name: Create local branch
+      command: ostree refs --create local-branch {{ head_csum }}
+
+    - name: Update origin file
+      command:  sed -i 's/^\(.*refspec\)=.*$/\1=local-branch/g' {{ current_dir.stdout }}.origin
```

Comment:
@jlebon I ran into an error running rpm-ostree reload yesterday.  I was going to ask you about it today since you were in training but now I can't reproduce it.  I've added the reload after modifying the origin file.